analytics_dashboard/static/apps/learners/js/learners-app.js

```
@@ -38,22 +51,23 @@ define([
         },
 
         onStart: function () {
-            // TODO: remove this temporary UI with AN-6205.
-            var LearnerView = Marionette.ItemView.extend({
-                template: _.template(
-                    '<div>' +
-                        '| <%- name %> | ' +
-                        '<%- username %> |' +
-                        '</div>'
-                )
-            }), LearnersView = Marionette.CollectionView.extend({
-                childView: LearnerView
+            var rootView = new LearnersRootView({el: $(this.options.containerSelector)}).render(),
+                router;
+            // Initialize our router and start keeping track of history
+            router = new LearnersRouter({
+                controller: new LearnersController({
+                    learnerCollection: this.learnerCollection,
+                    rootView: rootView
+                }),
+                learnerCollection: this.learnerCollection
             });
 
-            new LearnersView({
-                collection: this.learnerCollection,
-                el: $(this.options.containerSelector)
-            }).render();
+            try {
+                Backbone.history.start();
+            } catch (e) {
+                // If Backbone.history.start() has already been called,
```

IMO there is a cost in that potential exceptions are swallowed to handle a situation that shouldn't occur. I recommend removing it and fixing the tests that I think would break. In general, I see swallowing exceptions as being a bad code smell.